website/static/js/components/autocomplete.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Components involving the typeahead.js autocomplete library.
+ *
+ * Inspired by the onboarder component, but refactored into a generic view-model
+ * that can be subclassed to create components to search through lists of
+ * objects other than nodes.
+**/
+'use strict';
+require('css/autocomplete.css');
+
+var $ = require('jquery');
+var ko = require('knockout');
+var $osf = require('js/osfHelpers');
+require('typeahead.js');
+
+var MethodNotDefined = function(methodName) {
+    this.name = 'MethodNotDefined';
+    this.message = methodName;
+};
+MethodNotDefined.prototype = Error.prototype;
+
+/**
+ * ViewModel for a generic typeahead search component.
+ *
+ * Template: osf-search in components/autocomplete.js
+ *
+ * Params:
+ *  data: the URL from which to fetch the items to be searched
+ *  inputElement: the element upon which to call .typeahead(). Injected during
+ *      component registration.
+ */
+var baseSearchViewModel = function (params) {
```

Comment:
Use `UpperCamelCase` for class names.

https://github.com/felixge/node-style-guide#use-uppercamelcase-for-class-names